WcaOnRails/app/models/competition.rb

```diff
@@ -21,12 +21,15 @@ class Competition < ActiveRecord::Base
   MAX_ID_LENGTH = 32
   MAX_NAME_LENGTH = 50
   validates :id, presence: true, uniqueness: true, length: { maximum: MAX_ID_LENGTH },
-                 format: { with: /\A[a-zA-Z0-9]+\Z/ }
+                 format: { with: /\A[a-zA-Z0-9]+\Z/ }, if: :name_valid_or_updating?
+  private def name_valid_or_updating?
+    self.persisted? || name.length <= MAX_NAME_LENGTH && name =~ ENDS_WITH_YEAR_RE
```

Comment:
Despite of the fact that `&&` is stronger, I feel like parentheses should go here for explicitness. If you think that's personal feeling and you don't think the same, just ignore this comment =)